File: entity-meta-generator-lib/src/main/java/com/loudsight/meta/serialization/transform/LongEntityTransform.java
package com.loudsight.meta.serialization.transform;

import com.loudsight.meta.serialization.EntityTransform;
import com.loudsight.meta.serialization.EntityType;

import java.util.Iterator;
import java.util.List;

public class LongEntityTransform extends EntityTransform<Long> {

    private static class LongEntityTransformHolder {
        private static final LongEntityTransform INSTANCE = new LongEntityTransform();
    }
    // global access point
    public static LongEntityTransform getInstance() {
        return LongEntityTransform.LongEntityTransformHolder.INSTANCE;
    }
    private LongEntityTransform() {
        super(EntityType.LONG, Long.class);
    }


    @Override public void serializeEntity( Long entity, List<Byte> bytes) {
        bytes.add(EntityType.LONG.getCode());
        writeLong(entity, bytes);
    }

    @Override public Long  deserializeEntity(Iterator<Byte> bytes) {
        return readLong(bytes);
    }
}

File: entity-meta-generator-lib/src/main/kotlin/com/loudsight/meta/serialization/transform/LongEntityTransform.kt
package com.loudsight.meta.serialization.transform

import com.loudsight.meta.serialization.EntityTransform
import com.loudsight.meta.serialization.EntityType

object LongEntityTransform : EntityTransform<Long>(EntityType.LONG, Long::class) {

    override fun serializeEntity(entity: Long, bytes: MutableList<Byte>) {
        bytes.add(EntityType.LONG.code)
        writeLong(entity, bytes)
    }

    override fun deserializeEntity(bytes: Iterator<Byte>): Long {
        return readLong(bytes)
    }


    override fun canTransform(entity: Any): Boolean {
        return entity is Long
    }

}